Sources/TokamakLint/TokenVisitor.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  TokenVisitor.swift
+//  SwiftSyntax
+//
+//  Created by Matvii Hodovaniuk on 4/2/19.
+//
+
+import Foundation
+import SwiftSyntax
+
+class TokenVisitor: SyntaxVisitor {
+  public var tree = [Node]()
+  public var current: Node?
+
+  var row = 0
+  var column = 0
+
+  override func visitPre(_ node: Syntax) {
+    var syntax = "\(type(of: node))"
+    if syntax.hasSuffix("Syntax") {
+      syntax = String(syntax.dropLast(6))
+    }
+
+    let node = Node(text: syntax)
+    node.range.startRow = row
+    node.range.startColumn = column
+    node.range.endRow = row
+    node.range.endColumn = column
+    if current == nil {
+      tree.append(node)
+    } else {
+      current?.add(node: node)
+    }
+    current = node
+  }
+
+  override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
+    current?.text = escapeHtmlSpecialCharacters(token.text)
```

Comment:
would any of the code in this function execute when `current` is `nil`? If not, this could be optimised as `guard let current = current else { return }`